docs/DOCUMENTATION_GUIDELINES.md

```diff
@@ -0,0 +1,227 @@
+///
+/// Panics if the input overflows.
+///
+/// # Complexity
+///
+/// Is implemented using some algorithm that yields complexity of O(1).
+// More efficiency can be achieved if we improve this via such and such.
+fn multiply_by_2(x: u32) -> u32 { .. }
+```
+
+They are both roughly conveying the same set of facts, but one is substantially kinder on the eye. Especially for traits and type that you can foresee will be seen/used a lot, try and write the better version!
+
+Similarly, make sure your comments are wrapped at 100 characters line-width (as defined by our [`rustfmt.toml`](../rustfmt.toml)), no **more and no less**! The more is fixed by `rustfmt` and our CI, but if you (some some unknown reason) wrap your lines at 59 characters, it will pass the CI, and it will not look good 🫣.
+
+[^1]: Those that help two pallets talk to each other.
+
+## Pallet Crates
+
+Everything above is related to non-pallet details. They might be relevant in both crates that are pallets, and non-pallet crates.
+
+The following is relevant to how to document parts of a crate that is a pallet.
+
+### Top Level Pallet Docs (`lib.rs`)
+
+for the top-level pallet docs, consider the following template:
+
+```
+//! # <Pallet Name>
+//!
+//! <single-liner about the pallet>.
+//!
+//! ## High-Level/End-User Details
+//!
+//! <a few paragraphs, focus on what external folks should know about the pallet>
+//!
+//! <The audience here is potentially non-coders who just want to know what this pallet does, not how it does it>
+//!
+//! ### Example
+//!
+//! <Your pallet must have a few tests that cover important user journeys. Use https://crates.io/crates/docify to reuse your these as examples>.
+//!
+//! ## Code Details
+//!
+//! <inside [`pallet`] module, a template is auto-generated that leads the reader to the relevant items. This is just a reminder to navigate to that page. Don't repeat things like "See Config trait for ..." that are generated inside [`pallet`] in here>
+//!
+//! See [`pallet`] module.
+//!
+//! <The audience of this is those who want to know how this pallet works, to the extent of being able to build something on top of it, like a DApp or another pallet>
+//!
+//! ## Low Level / Implementation Details
+//!
+//! <The format of this section is up to you, but we suggest the Design-oriented approach that follows>
+//!
+//! <The audience of this would be your future self, and those who want to deeply understand the pallet to optimize it further and such>
+//!
+//! ### Design Goals (optional)
+//!
+//! <Describe your goals with with the pallet design.>
+//!
+//! ### Design (optional)
+//!
+//! <Describe how you reach those goals. This should namely describe how your storage is laid out.>
+//!
+//! ### Terminology (optional)
+//!
+//! <Optionally, contain any custom terminology here. You can link to it if you want to use the terminology further up>
+```
+
+### Dispatchable
+
+For each dispatchable (`fn` item inside `pallet::call`), consider the following template:
```

Comment:
```suggestion
For each dispatchable (i.e. the `fn` item inside `pallet_name::pallet::Call`), consider the following template:
```

Reply:
> `fn` item inside `pallet::call` 

is actually the correct location, not `Call`. Authors of pallets are not even exposed to `enum Call`.